mitxpro/views.py

```diff
@@ -45,6 +47,29 @@ def index(request, **kwargs):  # pylint: disable=unused-argument
     """
     if request.method == "GET":
         return render(request, "index.html", context=get_js_settings_context(request))
+    elif request.method == "POST":
```

Comment:
I don't think we should be doing a cookie-based approach here, we've had issues with cookie sizes impacting other services in weird ways and I've prefer to avoid writing a large object onto a cookie. Besides, I think this is more simply done by having the `DashboardPage` parse the query parameters out of what CyberSource is already sending when the page is loaded as a `POST`, sending the event off to GA, then doing a `history.go("/dashboard")` to clear the params. That's identical to what you're already doing here but doesn't use a cookie and keeps all this related login colocated in the same code.

Reply:
Updated, we simply need to embed the POST data from CyberSource into the page header (we're no longer just redirecting the POST request to a GET on the same URL, we're serving the page as usual with the POST data set in a variable along the same lines as our `SETTINGS` var. Upon sending the event to GA, reset that var to a `false` so this cycle is not repeated).